### DISCUSSION TITLE
UI/Component libraries known to work with Preact?

### DISCUSSION THREAD

**Discussion by IngwiePhoenix** [preactjs/preact#4559] [General]
Hello there!

Simple question out of curiosity, since most of the entries in `awesome-preact` are either outdated or outright archived: Are there any known good UI or component libriaries that work with Preact?

TailwindCSS technically does, but it's "just" CSS anyway. I'd love to not having to write a whole lot of components from scratch and most frameworks break sooner or later against preact/compat (I tried with schadcn-ui but Redux does not test their stuff against preact, so I ended up running into several issues and had to switch to React; making my bundle size explode from 60kb to 600...)

Thank you!

**Comment by rschristian** [comment]
> I tried with schadcn-ui but Redux does not test their stuff against preact

Do you mean Radix?

Some have gotten good use out of Shadn and/or Radix, though Radix specifically reaches quite deep into React internals and may not work in all cases (the React team has publicly stated what Radix is doing will break between React versions, FWIW). React95 I've also seen mention of. I think `@mui/material` generally works too.

I prefer Tailwind (well, [twind](https://npm.im/twind)) and writing components myself though, so that's just what I've heard from others.

**Reply by senpro-ingwersenk** [reply]
Oops, yes I ment Radix :) Radix, React, Redux... Resident Evil... to many REs!

Thank you for the pointers though, will give it a look!

**Comment by edwinwong90** [comment]
if you are looking shadcn like UI library, there are some of them (Chakra-ui and Park UI). But worth to mention that those UI library are using Ark-UI underlying (similar like Radix). What makes Ark-UI shines is because their framework agnostic solution which Preact are benefit from it.

My preference is using Ark-UI + DaisyUI as my UI library. Pretty great DX so far. Of course, It wont be as great as shadcn. That's the cost if you choose to using Preact. 

Noted that I don't use `preact/compact` at all. I want to keep my bundle size as small as possible.

**Comment by joshcancode** [comment]
With the latest shadcn update including Base UI variants of the components, almost all incompatibilities can be fixed with basic React aliasing and some minor tweaks. I made a boilerplate repo for Fresh (which uses Preact) which has every shadcn component updated for Preact with hardly any changes to the vanilla components.

**Comment by rafaell-lycan** [comment]
Not all but a few [Base UI](https://base-ui.com/) and [Ark UI](https://ark-ui.com/) if you do path aliasing in Vite:

```ts
    resolve: {
      alias: {
        'react': 'preact/compat',
        'react-dom': 'preact/compat',
      }
    }
```

You might get some type errors in a some cases.

**Reply by rschristian** [reply]
Please open an issue for the type errors if you have a chance, we'd love to fix them where we can

**Reply by EverStarck** [reply]
i was looking for this, thx!

**Comment by watzak** [comment]
https://github.com/kamod-ch/kamod-ui  (Preact component system / shadcn-style approach for Preact)
DEMO => https://kamod-ch.github.io/kamod-ui/
